crates/oxc_linter/src/utils/jest/parse_jest_fn_new.rs

```diff
@@ -0,0 +1,560 @@
+            {
+                return Err(ExpectError::ModifierUnknown);
+            }
+        } else {
+            return Err(ExpectError::ModifierUnknown);
+        }
+
+        modifiers.push(index);
+    }
+
+    Err(ExpectError::MatcherNotFound)
+}
+
+fn is_top_most_call_expr<'a, 'b>(node: &'b AstNode<'a>, ctx: &'b LintContext<'a>) -> bool {
+    let mut node = node;
+
+    loop {
+        let Some(parent) = ctx.nodes().parent_node(node.id()) else { return true };
+
+        match parent.kind() {
+            AstKind::CallExpression(_) => return false,
+            AstKind::MemberExpression(_) => node = parent,
+            _ => {
+                return true;
+            }
+        }
+    }
+}
+
+fn parse_jest_jest_fn_call<'a>(
+    members: Vec<KnownMemberExpressionProperty<'a>>,
+    name: &'a str,
+) -> Option<ParsedJestFnCall<'a>> {
+    if !name.to_ascii_lowercase().eq_ignore_ascii_case("jest") {
+        return None;
+    }
+
+    return Some(ParsedJestFnCall::GeneralJestFnCall(ParsedGeneralJestFnCall {
+        kind: JestFnKind::General(JestGeneralFnKind::Jest),
+        members,
+        name: Cow::Borrowed(name),
+    }));
+}
+
+#[derive(Clone, Copy, Debug)]
+pub enum ExpectError {
+    ModifierUnknown,
+    MatcherNotFound,
+    MatcherNotCalled,
+}
+
+pub struct ExpectFnCallOptions<'a, 'b> {
+    pub call_expr: &'a CallExpression<'a>,
+    pub members: Vec<KnownMemberExpressionProperty<'a>>,
+    pub name: &'a str,
+    pub head: KnownMemberExpressionProperty<'a>,
+    pub node: &'b AstNode<'a>,
+    pub ctx: &'b LintContext<'a>,
+}
+
+// If find a match in `VALID_JEST_FN_CALL_CHAINS`, return true.
+fn is_valid_jest_call(members: &[Cow<str>]) -> bool {
+    VALID_JEST_FN_CALL_CHAINS
+        .binary_search_by(|chain| {
+            chain
+                .iter()
+                .zip(members.iter())
+                .find_map(|(&chain, member)| {
+                    let ordering = chain.cmp(member.as_ref());
+                    if ordering != Ordering::Equal {
+                        return Some(ordering);
+                    }
+                    None
+                })
+                .unwrap_or(Ordering::Equal)
+        })
+        .is_ok()
+}
+
+fn resolve_to_jest_fn<'a>(
+    call_expr: &'a CallExpression<'a>,
+    original: Option<&'a Atom>,
+) -> Option<ResolvedJestFn<'a>> {
+    let ident = resolve_first_ident(&call_expr.callee)?;
+    Some(ResolvedJestFn { local: &ident.name, original })
+}
+
+fn resolve_first_ident<'a>(expr: &'a Expression) -> Option<&'a IdentifierReference> {
+    match expr {
+        Expression::Identifier(ident) => Some(ident),
+        Expression::MemberExpression(member_expr) => resolve_first_ident(member_expr.object()),
+        Expression::CallExpression(call_expr) => resolve_first_ident(&call_expr.callee),
+        Expression::TaggedTemplateExpression(tagged_expr) => resolve_first_ident(&tagged_expr.tag),
+        _ => None,
+    }
+}
```

Comment:
This file is copied from `parse_jest_fn.rs`, and I changed this function, others are just copied.